### PR TITLE
Time Complexity: fix link target and formatting

### LINKED ISSUE #26742: Time Complexity: Broken link to "Big Omega" knowledge check
### Describe your suggestion

The last link in the "Knowledge Check" section asks _"What is Big Omega and why isn’t it as useful?"_ The link points to `#big-937-omega-notation`, but there isn't anything with that `id` on the page. The closest thing is in the `section#alternatives-to-big-o`, where there's a `h4#big--omega-notation`. I'm guessing it's just a typo, and will be relatively easy to fix, but I wanted to ask if there's a reason behind the number `937` in the link. I don't see any in the other links, or indeed in any other `id`'s.

It looks like there was an attempt to fix this back in March with commit #25184, but that updating links from HTML to MD formatting in May with commit #25455 may have affected this.

Some quick research points to ampersands basically escape anything that comes after them, so the markdown header 
`#### Big &#937; (Omega Notation)` becomes the `id="big--omega-notation"`, effectively removing the number.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-time-complexity

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### (Optional) Discord Name

Marlatte

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by marlatte**
## Because

The last link in the "Knowledge Check" section asks _"What is Big Omega and why isn’t it as useful?"_ The current link points to `#big-937-omega-notation`, but there isn't anything with that 937 `id` on the page. However, there is a `h4#big--omega-notation`, where the 2 dashes come from a Unicode character being ignored when the lesson's markdown is converted into element id's for navigation. 

There were also several discrepancies between the JS and Ruby versions of the page.

## This PR

- Updated link in JS and Ruby .md files.
- Converted "Knowledge Check" links in Ruby version to md from html.
- Added link to "Additional resource" in JS version that was already present in Ruby version.
- 'Escaped' some `*` characters that Prettier and the MD previewer were misinterpreting as attempts to **bold** some words

## Issue

Closes #26742

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: javascript/computer_science/time_complexity.md
Comment (JoshDevHub) [inline javascript/computer_science/time_complexity.md:39]: This needs to go back to 2 space indenting instead of 8. Or these are possibly tabbed indents? In any case, we use 2 spaces to indent inside of codeblocks.

File: javascript/computer_science/time_complexity.md
Comment (JoshDevHub) [inline javascript/computer_science/time_complexity.md:78]: 2 space indent.

File: javascript/computer_science/time_complexity.md
Comment (JoshDevHub) [inline javascript/computer_science/time_complexity.md:247]: 2 space indent.

File: javascript/computer_science/time_complexity.md
Comment (JoshDevHub) [inline javascript/computer_science/time_complexity.md:287]: 2 space indent

File: javascript/computer_science/time_complexity.md
Comment (JoshDevHub) [inline javascript/computer_science/time_complexity.md:305]: 2 space indent.

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
Link to the "Omega" section looks good. Thank you for fixing it.

Have several formatting fixes that need to happen before I can merge. You should be able to quickly commit changes on all the list style ones. I wasn't able to suggest changes to the codeblock formatting, so you'll have to go in and add new commits that fix those yourself.

**Comment by marlatte** [comment]
Oops sorry about that. Looks like Prettier got carried away with formatting changes. Fixing now.

File: ruby/computer_science/time_complexity.md
Comment (marlatte) [reply ruby/computer_science/time_complexity.md:222]: Yes, Prettier and my MD previewer kept interpreting them as bold markers `*3*` instead of math symbols, so I escaped them.

File: javascript/computer_science/time_complexity.md
Comment (JoshDevHub) [inline javascript/computer_science/time_complexity.md:224]:
```suggestion
A factorial is the product of the sequence of _n_ integers. The factorial of 4(4!) is 4 * 3 * 2 * 1.
```

File: ruby/computer_science/time_complexity.md
Comment (JoshDevHub) [inline ruby/computer_science/time_complexity.md:222]:
```suggestion
A factorial is the product of the sequence of _n_ integers. For example, the factorial of 4 (written as 4!) is 4 * 3 * 2 * 1.
```

File: ruby/computer_science/time_complexity.md
Comment (JoshDevHub) [inline ruby/computer_science/time_complexity.md:226]:
```suggestion
The factorial of 3 is 6 (3 * 2 * 1). The factorial of 4 is 24. The factorial of 10? 3,628,800. So you can see how quickly things can get out of hand.
```

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
Alright awesome. Just going to request that we remove those escape characters before the `*`s. It doesn't impact the rendered output on the site, and we're not using it anywhere else in the curriculum. Good to stay consistent.

Also when you're previewing how Markdown content will be rendered on Odin's site, you can use our [markdown preview tool](https://www.theodinproject.com/lessons/preview). This will use the same parsing engine that's used for the lessons.

We should be good to merge after this change.

**Comment by JoshDevHub** [comment]
@marlatte Just reaching out to see if you're interested in getting this over the finish line. "No, I don't have time" is definitely an acceptable answer :smiley:

**Comment by marlatte** [comment]
Yes! Thanks for your patience with the delay.

**Review by JoshDevHub — APPROVED** [review]
It's no worries @marlatte 

We know people have busy lives, and this is all volunteer work. Thank you for your contribution :raised_hands:

**Pull request merged**
